pages/annotation.vue

```diff
@@ -29,22 +28,42 @@
 
             return {
 
-                mockupImage: {
+                // Full text name of this page
+				fullName: this.$store.getters.pageNames.categorization.fullName, 
 
+                // Image for temporary annotation page mockup
+                mockupImage: {
                     filename: "annotation_mockup.jpg",
                     height: 956,
                     width: 1045
-                }
+                },
+
+                // Initial status of the navbar items for other pages
+                navItemsState: [
```

Comment:
In fact, the navbar is not a shared thing. It just looks that way. Because each page is its own template and not inheriting from something like we were doing in Django, the navbar is created for each page with knowledge of what it should display (at least in its initial state).

I played around with exactly a implementation that saved navbar state per page in the store. (Thus the checkPageAccessibility method you came across.) Since knowledge of each pages navbar isn't needed by other pages I came to the conclusion that info on it shouldn't be in the store.

What does need polishing/deciding though is something else I played around with: whether or not the name of the current page should be on the navbar. I removed it for two reasons: 

1) It's more difficult to visually distinguish between  enabled, disabled, and active nav items
2) The page name can alternatively (and more prominently) be displayed as part of the navbar brand, programmatically via a prop as it is being done now.

Obviously, listing it twice would be pointless. With the current solution, I believe I solved the difficulty in visually understanding what was happening in the nav with enabled items, disabled items, and the active item.

Let me know what you think.

Reply:
Yeah, I think this is a really important point, thanks for doing this! My personal opinion here would be to keep things consistent with the bootstrap visual language. They have a way to differentiate "active", "selectable", and "disabled" and while that may not be the most visually clear, it's probably a more common and expected style - and that probably helps with intuitive understandability. So my vote would be for a single navbar where we style the current tab (and disabled tabs) using the bootstrap style.